hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -210,4 +236,84 @@ protected static function clearEffectiveEndDate($revisionId) {
     );
     CRM_Core_DAO::executeQuery($clearEffectiveEndDateQuery, $clearEffectiveEndDateParams);
   }
+
+  /**
+   * Builds the SQL query for fetching the revision details of the given entities
+   *
+   * The fields of each entities are dynamically listed in the SELECT clause
+   * and are aliased with a prefix ({entity}__) so that it is possible to know
+   * which entity they belong to (civicrm_hrjobcontract_pay.id -> pay__id)
+   *
+   * @param Array $entities
+   * @param Array $revision
+   */
+  protected static function buildFullDetailsQuery($entities, $revision) {
+    $query = ['select' => [], 'from' => [], 'where' => []];
+
+    /**
+     * Creates an alias for the given entity
+     *
+     * @param string $entity
+     * @return  string
+     */
+    $entityAlias = function ($entity) {
```

Comment:
I think you don't need this entityAlias function at all since you can just use the entity as the alias. What do you think?
Besides, the way it's was implemented, it can cause problems in the future. What if we need to add a new entity which starts with the same 3 letters as one of those entities?